programs/utilities/block_counter.sc

```diff
@@ -46,4 +165,180 @@ dpad(num, wid) ->
    strn = str(num);
    if (length(strn) < wid, strn = '...'*(wid-length(strn))+strn);
    strn;
-)
\ No newline at end of file
+);
+
+hpad(num, wid) ->
+(
+   strn = str(num);
+   if (length(strn) < wid, strn = strn+'...'*(wid-length(strn)));
+   strn;
+);
+
+vertical_report() -> (
+   //generate a map of the y level that contains each block
+   stats = global_data:'stats';
+   area = global_data:'area';
+   blocks = global_data:'blocks';
+
+   if(blocks=={},
+      for(reverse(sort(keys(stats))),
+         y =  _; level_stats = stats:y;
+         for(keys(level_stats),
+            if(blocks:_==null,
+               blocks:_ = [y],
+               blocks:_ += y
+            )
+         )
+      )
+   );
+   if(blocks=={}, nothing_found());
+
+   //print block names and y levels as clickable things to display tallys and histograms
+   print(format(make_clickable(blocks, 'histogram', 'y')));
+   print(format(make_clickable(stats, 'tally_level', 'c')));
+   print(format('m [Get book report]', str('!/%s book', system_info('app_name')) ));
+);
+
+make_clickable(input_map, fun, color) -> (
+   output = [];
+   for(sort(keys(input_map)),
+      if(input_map:_=={}, continue());
+      line = [
+         str('%s [%s] ', color, _), 
+         '^g Click to display results for '+_,
+         str('!/%s %s %s', system_info('app_name'), fun, _)
+      ];
+      put(output, null, line, 'extend');
+   );
+   output
+);
+
+histogram(block) -> (
```

Comment:
Finer histogram code if you want to put it in
```
histogram(block) -> (
   block = str(block);
   blocks = global_data:'blocks';
   block_characters = ['','▏','▎','▍','▌','▋','▊','▉','█'];
   divisions = length(block_characters) - 1;

   if(blocks:block==null, _error(block + ' not available from last scan.'));
   maxcount = 0;
   values = map(blocks:block,
      y = _;
      count = global_data:'stats':y:block;
      maxcount = max(maxcount, count);
      [y, count];
   );
   divisor = ceil(maxcount/30); //to make sure the histogram bar is not longer than the chat box
   print('');
   print(format('y Results for ', 'yb '+block));
   for(values, 
      [y, count] = _;
      ratio = count / divisor;
      print(format(str('w %s %s%s', hpad(y, 4), floor(ratio) * block_characters:(-1),block_characters:round(divisions * (ratio % 1))), '^g count: '+count) )
   );
);
```

Reply:
Not sure which one I prefer. What do people think? New histogram is more precise, but idk if that precision is needed or useful. You get the exact value if you hover over the bar, btw.

Current look of histogram is
![image](https://user-images.githubusercontent.com/64820010/146222373-1e5e2f81-b8e9-4563-ae8a-0bf50533f5b8.png)

and suggested new look (with better precision) is
![image](https://user-images.githubusercontent.com/64820010/146222262-ed70072e-46f5-44a6-8299-7b73760d519e.png)

Reply:
Just for reference, ｜(fullwidth vertical line, \uFF5C) is thinner than | (vertical line or pipe, \u007C) and offers even more resolution than █ (full block \u2588)

Reply:
hmm. Personally I prefer rv3r's solution